pkg/controllerhelpers/sync.go

```diff
@@ -37,20 +37,28 @@ func (f ControlleeManagerGetObjectsFuncs[CT, T]) PatchObject(ctx context.Context
 
 var _ ControlleeManagerGetObjectsInterface[kubeinterfaces.ObjectInterface, kubeinterfaces.ObjectInterface] = ControlleeManagerGetObjectsFuncs[kubeinterfaces.ObjectInterface, kubeinterfaces.ObjectInterface]{}
 
-func GetObjects[CT, T kubeinterfaces.ObjectInterface](
+func GetObjectsWithFilter[CT, T kubeinterfaces.ObjectInterface](
 	ctx context.Context,
 	controller metav1.Object,
 	controllerGVK schema.GroupVersionKind,
 	selector labels.Selector,
+	filterFunc func(T) bool,
 	control ControlleeManagerGetObjectsInterface[CT, T],
 ) (map[string]T, error) {
 	// List all objects to find even those that no longer match our selector.
 	// They will be orphaned in ClaimObjects().
-	objects, err := control.ListObjects(labels.Everything())
+	allObjects, err := control.ListObjects(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 
+	var objects []T
+	for i := range allObjects {
+		if !filterFunc(allObjects[i]) {
```

Comment:
Shouldn't it append objects matching the filter? Doing the opposite is usually called filtering **out**.
Say if you have a set of numbers, if you apply isEven filter, then as a result you get even numbers. 
In this controller we want to select every job matching a selector, and running on the same host. Now we are selecting these Jobs that are matching the selector, then filter is checking whether it's not running on the node, and thanks to double negation, we get jobs running on the same node.

Reply:
it should have ;)

Reply:
i even think the previous version that got dropped did this. i guess that what happens while you have to code over while having lunch :D